### PR TITLE
Appointment Model

### PR DESCRIPTION
**Pull request by ilaflamme-va**

### CODE REVIEW

File: argonaut-api/src/main/java/gov/va/api/health/argonaut/api/resources/Appointment.java [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+  @Pattern(regexp = Fhir.CODE)
+  String language;
+
+  @Valid Narrative text;
+  @Valid List<SimpleResource> contained;
+  @Valid List<Extension> extension;
+  @Valid List<Extension> modifierExtension;
+  @Valid List<Identifier> identifier;
+
+  Status status;
```
Comment (esubocz-va) [inline argonaut-api/src/main/java/gov/va/api/health/argonaut/api/resources/Appointment.java]: status is required. should be @NotBlank

File: argonaut-api/src/main/java/gov/va/api/health/argonaut/api/resources/Appointment.java [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+  String description;
+
+  @Pattern(regexp = Fhir.INSTANT)
+  String start;
+
+  @Pattern(regexp = Fhir.INSTANT)
+  String end;
+
+  int minutesDuration;
```
Comment (esubocz-va) [inline argonaut-api/src/main/java/gov/va/api/health/argonaut/api/resources/Appointment.java]: positive int, should it have @Min ?

File: argonaut-api/src/main/java/gov/va/api/health/argonaut/api/resources/Appointment.java [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+
+  @Valid Reference slot;
```
Comment (esubocz-va) [inline argonaut-api/src/main/java/gov/va/api/health/argonaut/api/resources/Appointment.java]: This should be a list according to spec. This change will require an edit to the sample data as well.

File: argonaut-api/src/main/java/gov/va/api/health/argonaut/api/resources/Appointment.java (hunk 2 above)
Comment (bschofield-va) [reply argonaut-api/src/main/java/gov/va/api/health/argonaut/api/resources/Appointment.java]: It's also optional, should be Integer.

**Pull request merged**